opm/material/fluidmatrixinteractions/EclMaterialLawManager.hpp

```diff
@@ -346,8 +358,76 @@ class EclMaterialLawManager
 
             // read the end point scaling info for the saturation region
             unscaledEpsInfo_[satnumIdx].extractUnscaled(deck, eclState, satnumIdx);
+        }
+
+        EclEpsGridProperties epsGridProperties, epsImbGridProperties;
+        epsGridProperties.initFromDeck(deck, eclState, /*imbibition=*/false);
+        if (enableHysteresis())
+            epsImbGridProperties.initFromDeck(deck, eclState, /*imbibition=*/true);
+
+        initParamsForElements_(materialLawParams_, deck, eclState, compressedToCartesianElemIdx, satnumRegionArray,
+                               gasOilUnscaledPointsVector, oilWaterUnscaledPointsVector, gasOilEffectiveParamVector,oilWaterEffectiveParamVector,
+                               epsGridProperties, epsImbGridProperties );
+
 
+        // for well connections
+        std::vector<int> satnumRegionArrayConnections;
+        std::vector<int> compressedToCartesianElemIdxConnections;
+
+        const auto& schedule = eclState.getSchedule();
+        const auto& eclGrid = eclState.getInputGrid();
+
+        size_t timeStep = 0;
+        auto wells = schedule.getWells(timeStep); //need all wells ...
```

Comment:
The `struct Wells` is of @joakim-hove's first kind (should be immutable), and `WellState` is of the second kind. The mutable information (for hysteresis etc.) related to each well completion should go in the WellState, and definitely not in the opm-parser Completion class (which is of the first kind, representing "what was given in the deck" without any regard to "what is the current state").